File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;

int main(void){
  ios::sync_with_stdio(0);
  cin.tie(0);
  
  int n;
  cin >> n;

  stack<int> s;
  for(int i=0;i<n;i++){
    string command;
    cin >> command;
    if(command == "push"){
      int num;
      cin >> num;
      s.push(num);
    } else if (command == "pop"){
      if(s.empty()){
        cout << -1 << '\n';
      } else {
        cout << s.top() << '\n';
        s.pop();
      }
    } else if (command == "top"){
      if(s.empty()){
        cout << -1 << '\n';
      } else {
        cout << s.top() << '\n';
      }
    } else if (command == "size"){
      cout << s.size() << '\n';
    } else if (command == "empty"){
      if(s.empty()){
        cout << 1 << '\n';
      } else {
        cout << 0 << '\n';
      }
    }
  }

  return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = null;

    int n = Integer.parseInt(br.readLine());
    Stack<Integer> stack = new Stack<>();

    for(int i=0; i<n; i++){
      st = new StringTokenizer(br.readLine());
      String command = st.nextToken();

      if(command.equals("push")){
        int num = Integer.parseInt(st.nextToken());

        stack.push(num);
      } else if (command.equals("pop")){
        if(stack.isEmpty()){
          System.out.println(-1);
        } else {
          int popNum = stack.pop();
          System.out.println(popNum);
        }
      } else if (command.equals("size")){
        System.out.println(stack.size());
      } else if (command.equals("empty")){
        if (stack.isEmpty()) {
          System.out.println(1);
        } else {
          System.out.println(0);
        }
      } else if (command.equals("top")){
        if(stack.isEmpty()){
          System.out.println(-1);
        } else {
          System.out.println(stack.peek());
        }
      }
    }
  }
}
